File: src/main/java/org/blacksun/graph/channel/DuplexChannelFactory.java
package org.blacksun.graph.channel;

import org.blacksun.graph.node.GraphNode;
import org.jetbrains.annotations.NotNull;

public class DuplexChannelFactory implements ChannelFactory {
    @Override
    public Channel createChannel(@NotNull GraphNode from, @NotNull GraphNode to, int weight, double errors) {
        return new DuplexChannel(from, to, weight, errors);
    }

    @Override
    public String toString() {
        return "Duplex";
    }
}

File: src/main/java/org/blacksun/graph/channel/DuplexChannelFactory.kt
package org.blacksun.graph.channel

import org.blacksun.graph.node.GraphNode

class DuplexChannelFactory : ChannelFactory {
    override fun createChannel(from: GraphNode, to: GraphNode, weight: Int, errors: Double): Channel {
        return DuplexChannel(from, to, weight, errors)
    }

    override fun toString(): String {
        return "Duplex"
    }
}
